### PR TITLE
Add tests to firehose transformer

### PR DESCRIPTION
**Pull request by TomPretty**
## What are you doing in this PR?
Add tests to firehose transformer.

### CODE REVIEW

File: support-lambdas/acquisitions-firehose-transformer/src/main/scala/com/gu/acquisitionFirehoseTransformer/Lambda.scala [hunk 1]
```diff
@@ -32,7 +32,11 @@ case class AcquisitionWithRecordId(val acquisition: AcquisitionDataRow, recordId
 
 object Lambda extends LazyLogging {
 
-  def handler(event: KinesisFirehoseEvent, context: Context): KinesisAnalyticsInputPreprocessingResponse = {
+  def handler(event: KinesisFirehoseEvent): KinesisAnalyticsInputPreprocessingResponse = {
+    processEvent(event, new AnnualisedValueServiceImpl())
+  }
+
+  def processEvent(event: KinesisFirehoseEvent, avService: AnnualisedValueServiceWrapper) = {
```
Comment (TomPretty) [inline support-lambdas/acquisitions-firehose-transformer/src/main/scala/com/gu/acquisitionFirehoseTransformer/Lambda.scala:39]: We've factored out a `processEvent` method so we can make use of dependency injection in our tests to provide a simple mock av service

File: support-lambdas/acquisitions-firehose-transformer/src/test/scala/com/gu/acquisitionFirehoseTransformer/LambdaSpec.scala [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import org.joda.time.DateTime
+import com.gu.i18n.{Country, Currency}
+import com.gu.support.zuora.api.ReaderType
+import java.nio.ByteBuffer
+import collection.JavaConverters._
+import com.gu.acquisitionsValueCalculatorClient.model.AcquisitionModel
+import scala.concurrent.ExecutionContext
+
+
+class LambdaSpec extends AnyFlatSpec with Matchers {
```
Comment (tomrf1) [inline support-lambdas/acquisitions-firehose-transformer/src/test/scala/com/gu/acquisitionFirehoseTransformer/LambdaSpec.scala:19]: 💯 shall we delete the old test file now? I think the new tests cover serialization just as well

**Review by tomrf1 — APPROVED** [review]
👍 great!

File: support-lambdas/acquisitions-firehose-transformer/src/test/scala/com/gu/acquisitionFirehoseTransformer/LambdaSpec.scala (hunk 2 above)
Comment (TomPretty) [reply support-lambdas/acquisitions-firehose-transformer/src/test/scala/com/gu/acquisitionFirehoseTransformer/LambdaSpec.scala:19]: yeah makes sense - I'll remove that now! 👍

**Pull request merged**